internal/common/baseclient/iito.go

```diff
@@ -2,43 +2,53 @@ package baseclient
 
 import "github.com/SOMAS2020/SOMAS2020/internal/common/shared"
 
-// RequestGift allows clients to signalize that they want a gift
+// FIXME: Wait for implementation of getClientsAlive()
+const aliveClients = 6
+
+// GetGiftRequests allows clients to signalize that they want a gift
 // This information is fed to OfferGifts of all other clients.
 // COMPULSORY, you need to implement this method
-func (c *BaseClient) RequestGift() uint {
-	if c.clientGameState.ClientInfo.LifeStatus == shared.Critical {
-		return 100
+func (c *BaseClient) GetGiftRequests() shared.GiftRequestDict {
+	requests := make(shared.GiftRequestDict, aliveClients)
+	for key := range requests {
+		if c.clientGameState.ClientInfo.LifeStatus == shared.Critical {
+			requests[key] = shared.GiftRequest{RequestingTeam: c.GetID(), OfferingTeam: key, RequestAmount: 100.0}
```

Comment:
Clients should not be allowed to specify the requesting team since they could pretend to be someone else.